src/OpenTelemetry/Metrics/Aggregator/LastValueAggregator.cs

```diff
@@ -23,31 +23,48 @@ namespace OpenTelemetry.Metrics
 {
     public class LastValueAggregator : Aggregator
     {
-        private readonly KeyValuePair<string, object>[] emptyKVP = new KeyValuePair<string, object>[0];
         private readonly Instrument instrument;
-        private readonly string[] names;
-        private readonly object[] values;
+        private readonly KeyValuePair<string, object>[] tags;
 
-        private IDataPoint lastValue;
+        private readonly object lockLastValue = new object();
         private int count = 0;
+        private DataPoint lastDataPoint;
 
         public LastValueAggregator(Instrument instrument, string[] names, object[] values)
         {
+            this.instrument = instrument;
+
             if (names.Length != values.Length)
             {
                 throw new ArgumentException("Length of names[] and values[] must match.");
             }
 
-            this.instrument = instrument;
-            this.names = names;
-            this.values = values;
+            this.tags = new KeyValuePair<string, object>[names.Length];
+            for (int i = 0; i < names.Length; i++)
+            {
+                this.tags[i] = new KeyValuePair<string, object>(names[i], values[i]);
+            }
         }
 
         public override void Update<T>(DateTimeOffset dt, T value)
             where T : struct
         {
-            this.count++;
-            this.lastValue = new DataPoint<T>(dt, value, this.emptyKVP);
+            lock (this.lockLastValue)
+            {
+                this.count++;
+                if (typeof(T) == typeof(int))
```

Comment:
Probably faster to use `is` here. `if (value is int intValue)` type of deal.

Reply:
Problem with "is" is that on one framework, it allocates memory!

Reply:
That is right. The JIT on the full framework can cause boxing allocation.

Reply:
Good to know!